src/table/pagination-table/components/styles.ts

```diff
@@ -27,12 +27,15 @@ export const StyledTableContainer = styled(TableContainer, {
 }));
 
 export const StyledTable = styled(Table, {
-  shouldForwardProp: (prop: string) => prop !== 'customWidth',
-})(({ customWidth }) =>
+  shouldForwardProp: (prop: string) => prop !== 'customWidth' && prop !== 'isBorderRight' && prop !== 'styling',
+})(({ customWidth, isBorderRight, styling }) =>
   customWidth
     ? {
         tableLayout: 'fixed',
         width: 'min-content',
+        borderWidth: isBorderRight && '0px 1px 0px 0px',
+        borderStyle: isBorderRight && 'solid',
+        borderColor: isBorderRight && styling.body.borderRightColor,
```

Comment:
you could do this in one line with right-border
```suggestion
        borderRight: isBorderRight && `1px solid ${styling.body.borderRightColor}`,
```